src/v0/destinations/fb_custom_audience/util.js

```diff
@@ -221,9 +227,44 @@ const generateAppSecretProof = (accessToken, appSecret, dateNow) => {
   return appsecretProof;
 };
 
+const getDataSource = (type, subType) => {
+  const dataSource = {};
+  if (type && type !== 'NA' && typeFields.includes(type)) {
+    dataSource.type = type;
+  }
+  if (subType && subType !== 'NA' && subTypeFields.includes(subType)) {
+    dataSource.sub_type = subType;
+  }
+  return dataSource;
+};
+
+const responseBuilderSimple = (payload, audienceId) => {
+  if (payload) {
+    const responseParams = payload.responseField;
+    const response = defaultRequestConfig();
+    response.endpoint = getEndPoint(audienceId);
+
+    if (payload.operationCategory === 'add') {
+      response.method = defaultPostRequestConfig.requestMethod;
+    }
+    if (payload.operationCategory === 'remove') {
+      response.method = defaultDeleteRequestConfig.requestMethod;
+    }
+
+    response.params = responseParams;
+    return response;
+  }
+  // fail-safety for developer error
+  throw new TransformationError(`Payload could not be constructed`);
+};
+
 module.exports = {
   prepareDataField,
   getSchemaForEventMappedToDest,
   batchingWithPayloadSize,
```

Comment:
@ItsSudip Were these checks ensureApplicableFormat & getUpdatedDataElement missing in the old audienceList flow ?

Reply:
@krishna2020 these are also present in audience flow, they are called with `prepareDataField` in transform.js